### PR TITLE
Add SQL and item scripts for Scroll of Instant Protect and Shell

### PR DESCRIPTION
**Pull request by CatsEyeXI**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

Currently the Scrolls of Instant Protect and Shell (purchased from Curio Vendor Moogles) cannot be activated. This enables them to be used.  Also updated animation for scroll of stoneskin based on retail cap.

```
[5988] = { ['name'] = "Instant Protect",   ['animation'] = 30, },
[5989] = { ['name'] = "Instant Shell",     ['animation'] = 32, },
[5990] = { ['name'] = "Instant Stoneskin", ['animation'] =  8, },
```

## Steps to test these changes

!additem 5988
!additem 5989

Use both

### CODE REVIEW

**Comment by WinterSolstice8** [comment]
It is unclear what power these new luas are using. Please verify with retail caps in case SE fixed it for the case of Instant Protect or use the values provided by bg wiki.

https://www.bg-wiki.com/ffxi/Instant_Protect
https://www.bg-wiki.com/ffxi/Instant_Shell

File: scripts/globals/items/scroll_of_instant_protect.lua [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+local item_object = {}
+
+item_object.onItemCheck = function(target)
+    return 0
+end
+
+item_object.onItemUse = function(target)
+    local power = 1
+    local duration = 1800
+    if (target:hasStatusEffect(xi.effect.PROTECT)) then
```
Comment (zach2good) [inline scripts/globals/items/scroll_of_instant_protect.lua]: https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md#no-parentheses-unless-needed-to-clarify-order-of-operations

File: scripts/globals/items/scroll_of_instant_protect.lua [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        local effect = target:getStatusEffect(xi.effect.PROTECT)
+        local oPower = effect:getPower()
+        if (oPower > power) then
```
Comment (zach2good) [inline scripts/globals/items/scroll_of_instant_protect.lua]: https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md#no-parentheses-unless-needed-to-clarify-order-of-operations

File: scripts/globals/items/scroll_of_instant_shell.lua [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+local item_object = {}
+
+item_object.onItemCheck = function(target)
+    return 0
+end
+
+item_object.onItemUse = function(target)
+    local power = 1
+    local duration = 1800
+    if (target:hasStatusEffect(xi.effect.SHELL)) then
```
Comment (zach2good) [inline scripts/globals/items/scroll_of_instant_shell.lua]: https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md#no-parentheses-unless-needed-to-clarify-order-of-operations

File: scripts/globals/items/scroll_of_instant_shell.lua [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        local effect = target:getStatusEffect(xi.effect.SHELL)
+        local oPower = effect:getPower()
+        if (oPower > power) then
```
Comment (zach2good) [inline scripts/globals/items/scroll_of_instant_shell.lua]: https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md#no-parentheses-unless-needed-to-clarify-order-of-operations

File: scripts/globals/items/scroll_of_instant_protect.lua [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+require("scripts/globals/msg")
+-----------------------------------
+local item_object = {}
+
+item_object.onItemCheck = function(target)
+    return 0
+end
+
+item_object.onItemUse = function(target)
+    local power = 1
```
Comment (zach2good) [inline scripts/globals/items/scroll_of_instant_protect.lua]: Like @WinterSolstice8 says, please get caps from retail for these values.

File: scripts/globals/items/scroll_of_instant_shell.lua [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+require("scripts/globals/msg")
+-----------------------------------
+local item_object = {}
+
+item_object.onItemCheck = function(target)
+    return 0
+end
+
+item_object.onItemUse = function(target)
+    local power = 1
```
Comment (zach2good) [inline scripts/globals/items/scroll_of_instant_shell.lua]: Like @WinterSolstice8 says, please get caps from retail for these values.

**Comment by CatsEyeXI** [comment]
Protect:
image

**Comment by CatsEyeXI** [comment]
I guess there's no easy way to find out what Shell gives for magic defense. I'll use the BG Wiki provided value if that's alright

**Comment by TeoTwawki** [comment]
> Protect: image

now that you've down prot should not be 1 defense, figure out if it scales like fov book protect does. and then shell is next.

Anything that does a normally fixed amount of magic damage can show you shell's potency.

**Comment by TeoTwawki** [comment]
> I guess there's no easy way to find out what Shell gives for magic defense. I'll use the BG Wiki provided value if that's alright

Pretty simple actually. But if Bg has values for either/both of these that's good enough

File: scripts/globals/items/scroll_of_instant_protect.lua [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    local duration = 1800
+    if (target:hasStatusEffect(xi.effect.PROTECT)) then
+        local effect = target:getStatusEffect(xi.effect.PROTECT)
+        local oPower = effect:getPower()
+        if (oPower > power) then
+            target:messageBasic(xi.msg.basic.NO_EFFECT)
+        else
+            target:delStatusEffect(xi.effect.PROTECT)
```
Comment (TeoTwawki) [inline scripts/globals/items/scroll_of_instant_protect.lua]: ~~unless this overwrites protect effects _stronger_ than what this item provides, this should not be here. same for shell. Thats very unlikely and can be tested by casting prot 5 and then using the scroll, checking your defense before and after each (3 checks)~~

it silently fails if a stronger protect power is already on the player, so we need to both not delete the effect, and suppress the "no effect" message here.

**Comment by CatsEyeXI** [comment]
Sorry I misread, give me a bit.

**Comment by CatsEyeXI** [comment]
> > Protect: image
> 
> now that you've down prot should not be 1 defense, figure out if it scales like fov book protect does. and then shell is next.
> 
> Anything that does a normally fixed amount of magic damage can show you shell's potency.

It does not appear that the instant scroll  scales with your level. DEF +75 at both Lv99 and at Lv5
image

**Comment by TeoTwawki** [comment]
> It does not appear that the instant scroll scales with your level. DEF +75 at both Lv99 and at Lv5

[will it overwrite a stronger one, or fail to apply (no effect) or prevent use of the scroll?](https://github.com/LandSandBoat/server/pull/2406#discussion_r931542573)

**Comment by CatsEyeXI** [comment]
> > It does not appear that the instant scroll scales with your level. DEF +75 at both Lv99 and at Lv5
> 
> [will it overwrite a stronger one, or fail to apply (no effect) or prevent use of the scroll?](https://github.com/LandSandBoat/server/pull/2406#discussion_r931542573)

does not appear it does, no message returned at all.

image

**Comment by TeoTwawki** [comment]
> > > It does not appear that the instant scroll scales with your level. DEF +75 at both Lv99 and at Lv5
> > 
> > 
> > [will it overwrite a stronger one, or fail to apply (no effect) or prevent use of the scroll?](https://github.com/LandSandBoat/server/pull/2406#discussion_r931542573)
> 
> does not appear it does, no message returned at all.
> 
> image

since defense is the same before and after in your pic, its either a silent fail (your prot 5 defense is still in play) or your chatlog is filtering out the fail message

or its equal power, overwrite on equal or greater

**Comment by TeoTwawki** [comment]
or its equal power, overwrite on equal or greater

**Comment by TeoTwawki** [comment]
verified myself: silent failure, no overwrite of stronger effect.

File: scripts/globals/items/scroll_of_instant_protect.lua (hunk 1 above)
Comment (CatsEyeXI) [reply scripts/globals/items/scroll_of_instant_protect.lua]: sorry for missing this

**Comment by CatsEyeXI** [comment]
I'm going to try to iron out the correct animations and will update.

File: scripts/globals/items/scroll_of_instant_shell.lua [hunk 8]
```diff
@@ -13,20 +12,10 @@ item_object.onItemCheck = function(target)
 end
 
 item_object.onItemUse = function(target)
-    local power = 1
+    local power = 2188 -- shell_iii base power
```
Comment (TeoTwawki) [inline scripts/globals/items/scroll_of_instant_shell.lua]: Like I was saying on discord, bg says it uses the old pre-adjustment shell 3 power. Shell 3 used to be 19% (48/255) before being moved to its modern 22% (56/255)

when SE updated the spells, they left items alone (which is why the protect scroll is giving 75 and not the 90 that prot 3 does now)

Comment (CatsEyeXI) [reply scripts/globals/items/scroll_of_instant_shell.lua]: i understand now, sorry bout that

Comment (CatsEyeXI) [reply scripts/globals/items/scroll_of_instant_shell.lua]: so if the power is base 10000, 19% would be 1900, correct? :)

**Review by Xaver-DaRed — APPROVED** [review]

**Review by zach2good — APPROVED** [review]

**Pull request merged**